content/docs/testing-recipes.md

```diff
@@ -278,126 +220,106 @@ it("changes value when clicked", () => {
+
 ```jsx
 // card.test.js
-jest.useFakeTimers();
-
-let container = null
-beforeEach(() => {
-  // setup a DOM element as a render target
-  container = document.createElement('div')
-  document.body.appendChild(container)
-})
-
-afterEach(() => {
-  // cleanup on exiting
-  container.remove();
-  container = null  
-})
+import React from "react";
+import { render, fireEvent, act } from "@testing-library/react";
+import Card from "./card";
 
-function render(ui){
-  act(() => {
-    ReactDOM.render(ui, container);
-  });
-}
+jest.useFakeTimers();
 
-it("should select null after timing out", () => {
-  const yields = [];
-  render(<Card onSelect={selection => yields.push(selection)} />);
+test("selects null after timing out", () => {
+  const handleSelect = jest.fn();
+  render(<Card onSelect={handleSelect} />);
   act(() => {
     jest.advanceTimersByTime(100);
   });
-  expect(yields).toBe([]);
+  expect(handleSelect).not.toHaveBeenCalled();
   act(() => {
-    jest.advanceTimersByTime(1000);
+    jest.advanceTimersByTime(10000);
   });
-  expect(yields).toBe([null]);
+  expect(handleSelect).toHaveBeenCalledTimes(1);
+  expect(handleSelect).toHaveBeenCalledWith(null);
 });
 
-it("should cleanup on being removed", () => {
-  const yields = [];
-  render(
-    <Card onSelect={selection => yields.push(selection)} />
-  );
-  act(() => {
-    jest.advanceTimersByTime(100);
-  });
-  expect(yields).toBe([]);
+test("cleans up on being removed", () => {
+  const handleSelect = jest.fn();
+  const { unmount } = render(<Card onSelect={handleSelect} />);
+  unmount();
   act(() => {
-    render(null);
+    jest.advanceTimersByTime(10000);
   });
-  act(() => {
-    jest.advanceTimersByTime(1000);
-  });
-  expect(yields).toBe([]);
+  expect(handleSelect).not.toHaveBeenCalled();
 });
 
-it("should accept selections", () => {
-  const yields = [];
-  render(
-    <Card onSelect={selection => yields.push(selection)} />
-  );
+test("accepts selections", () => {
+  const handleSelect = jest.fn();
+  const { getByTestId } = render(<Card onSelect={handleSelect} />);
 
-  act(() => {
-    container
-      .querySelector("[data-test-id=2]")
-      .dispatchEvent(new MouseEvent("click", { bubbles: true }));
-  });
+  fireEvent.click(getByTestId("card-button-2"));
 
-  expect(yields).toBe([2]);
+  expect(handleSelect).toHaveBeenCalledTimes(1);
+  expect(handleSelect).toHaveBeenCalledWith(2);
 });
 ```
 
 ### Snapshot testing {#snapshot-testing}
 
-Frameworks like Jest also let you save 'snapshots' of data. Combined with react-test-renderer, we can save component structures, and verify they don't break in the future. It's typically better to make more specific assertions than to use snapshots (or at least ensure that the snapshot is small) because these kinds of tests include implementation details meaning they break easily and teams can get desensitized to snapshot breakages.
```

Comment:
Not using `react-test-renderer` because snapshots of that includes a TON of implementation details. Snapshotting the DOM isn't terrific either, but it doesn't include nearly as many implementation details. Therefore you'll get fewer false negatives.